# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1234945005 -3600
#      Wed Feb 18 09:16:45 2009 +0100
# Node ID 53f8fca6dd697055b1bfc5726578378463b15242
# Parent  ee3446ca6157d2b0eb78927b7225da9e6a49e1f6
fix typo in qr.cc

diff --git a/src/DLD-FUNCTIONS/qr.cc b/src/DLD-FUNCTIONS/qr.cc
--- a/src/DLD-FUNCTIONS/qr.cc
+++ b/src/DLD-FUNCTIONS/qr.cc
@@ -737,17 +737,17 @@ as\n\
 
 */
 
 static
 bool check_qr_dims (const octave_value& q, const octave_value& r,
                     bool allow_ecf = false)
 {
   octave_idx_type m = q.rows (), k = r.rows (), n = r.columns ();
-  return ((q.ndims () == 2 || r.ndims () == 2 && k == q.columns ())
+  return ((q.ndims () == 2 && r.ndims () == 2 && k == q.columns ())
             && (m == k || (allow_ecf && k == n && k < m)));
 }
 
 static 
 bool check_index (const octave_value& i, bool vector_allowed = false)
 {
   return ((i.is_real_type () || i.is_integer_type ()) 
           && (i.is_scalar_type () || vector_allowed));
